### PR TITLE
[CI] [GHA] Enable testing of Python 3.12 on Ubuntu 24

### PR DESCRIPTION
**Pull request by akashchi**
### Tickets:
 - *151255*

### CODE REVIEW

**Comment by akashchi** [comment]
@akuporos, please take a look at the skips, errors, and tickets. I think that if a fix is easy enough, it could be done via this PR, e.g., in case of a missing dependency: some test stages failed with things like `ModuleNotFoundError: No module named 'onnxruntime'/jax`.

File: .github/workflows/ubuntu_24.yml [hunk 1]
```diff
@@ -17,7 +17,6 @@ permissions: read-all
 
 env:
   PIP_CACHE_PATH: /mount/caches/pip/linux
-  PYTHON_VERSION: '3.11'
```
Comment (akashchi) [inline .github/workflows/ubuntu_24.yml:20]: Unfortunately, we can't use `${{ env.PYTHON_VERSION }}` as an input for a reusable workflow as the `env` context is not propagated to the reusable jobs, we can do a workaround like in [1](https://github.com/orgs/community/discussions/26671) but currently we only have one variable needing to be passed so we might have it as it is now.

File: .github/workflows/job_python_unit_tests.yml [hunk 2]
```diff
@@ -231,7 +234,7 @@ jobs:
           PYTORCH_TRACING_MODE: TORCHFX
 
       - name: ONNX Layer Tests
-        if: fromJSON(inputs.affected-components).ONNX_FE.test
+        if: ${{ fromJSON(inputs.affected-components).ONNX_FE.test && inputs.python-version != '3.12' }} # Ticket: 152240
```
Comment (ilya-lavrenov) [inline .github/workflows/job_python_unit_tests.yml:237]: looks like it's available https://pypi.org/project/onnxruntime/#files, but only for glibc 2.27 and newer

looks like we need to change the condition

Comment (akashchi) [reply .github/workflows/job_python_unit_tests.yml:237]: it seems like that it is installed in the Layer tests dependencies step: https://github.com/openvinotoolkit/openvino/actions/runs/10812194091/job/29994666552#step:16:117 but is not found during tests execution.

File: tools/mo/unit_tests/mo/ops/elementwise_test.py [hunk 3]
```diff
@@ -116,6 +121,10 @@ def test_first_input_const(self):
         const_node = Node(graph, 'const')
         self.assertEquals(const_node.out_port(0).get_data_type(), np.float32)
```
Comment (akuporos) [inline tools/mo/unit_tests/mo/ops/elementwise_test.py:122]:
```suggestion
        self.assertEqual(const_node.out_port(0).get_data_type(), np.float32)
```

Comment (akuporos) [inline tools/mo/unit_tests/mo/ops/elementwise_test.py:122]:
```suggestion
        self.assertEqual(const_node.out_port(0).get_data_type(), np.float32)
```

Comment (akuporos) [reply tools/mo/unit_tests/mo/ops/elementwise_test.py:122]: and in three other tests just need to change assertEquals -> assertEqual

File: tests/layer_tests/pytorch_tests/test_repeat.py [hunk 4]
```diff
@@ -82,5 +84,9 @@ def forward(self, x):
     @pytest.mark.precommit
     @pytest.mark.precommit_torch_export
     @pytest.mark.precommit_fx_backend
+    @pytest.mark.skipif(
+        condition=(sys.version_info[0], sys.version_info[1]) == (3, 12),
+        reason='Ticket: 152223'
```
[the remark this answers is not in the document]
Comment (akashchi) [reply tests/layer_tests/pytorch_tests/test_repeat.py:89]: The dependencies install step has `transformers` installed: https://github.com/openvinotoolkit/openvino/actions/runs/10812194091/job/29994666552#step:16:117, I could find the `transformers` dependency in the requirements files but it did not contain any restrictions for Python versions:
https://github.com/openvinotoolkit/openvino/blob/97030adcbd5e5d8cd27e7a9450fdba967943f18c/tests/layer_tests/requirements.txt#L11
and
https://github.com/openvinotoolkit/openvino/blob/97030adcbd5e5d8cd27e7a9450fdba967943f18c/tests/constraints.txt#L30

so the dependency is not seen in the tests for some reason.

File: tools/mo/unit_tests/mo/ops/elementwise_test.py (hunk 3 above)
Comment (akashchi) [reply tools/mo/unit_tests/mo/ops/elementwise_test.py]: Seems like it helped, thanks, I'll close the corresponding ticket.

File: tools/mo/unit_tests/mo/ops/elementwise_test.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Copyright (C) 2018-2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+import sys
```
Comment (akuporos) [inline tools/mo/unit_tests/mo/ops/elementwise_test.py:4]:
```suggestion
```

File: tests/layer_tests/pytorch_tests/test_repeat.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (akashchi) [reply tests/layer_tests/pytorch_tests/test_repeat.py:89]: Yeah, it did not install the dependencies on the failing run as some stages failed before that and I forgot to use `always()` for the install dependencies step.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by rkazants — APPROVED** [review]
I got it. It is good that 3.11 is left on Ubuntu 22 and other platforms where these tests are running.

**Pull request merged**